staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/validator.go

```diff
@@ -223,9 +228,31 @@ func (v *CELValidator) Validate(versionedAttr *generic.VersionedAttributes, vers
 			continue
 		}
 		t1 := time.Now()
-		evalResult, _, err := compilationResult.Program.Eval(va)
+		evalResult, evalDetails, err := compilationResult.Program.Eval(va)
 		elapsed := time.Since(t1)
 		policyDecision.Elapsed = elapsed
+		if evalDetails == nil {
+			policyDecision.Action = policyDecisionActionForError(f)
+			policyDecision.Evaluation = EvalError
+			policyDecision.Message = fmt.Sprintf("runtime cost could not be calculated for expression: %v, no further expression will be run", v.policy.Spec.Validations[i].Expression)
+			return decisions, nil
```

Comment:
Previously we agreed on saving single expression related error in the decision of current expression evaluation result. Since the error might not affect other expressions. I agree it should be the internal error but wondering if we should drop result from other expressions

Reply:
After offline discussion, the cost err is not expected at all especially after passed compilation. And we would love the way to save errs per expression. In this case we will keep on the current way of handling. I will keep the conversation open for couple more days in case.